### PR TITLE
Civ 5011 keyword changes update fee pay new

### PR DESCRIPTION
**Pull request by asthamalviya**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-5011

### Change description ###
Added fee and payment api classes which were before used from a jar. 
As per new change a new keyword will be passed as parameter.
This call is to calculate the fee and give us fee back.

As part of this change, we need to send an extra parameter in Fee&Pay api service call for both claim issuance and hearing fee.

Impacted modules: civil-money-claims

Current Fee&Pay API call is:

[http://fees-register-api-prod.service.core-compute-prod.internal:80/fees-register/fees/lookup?service=civil%20money%20claims&jurisdiction1=civil&jurisdiction2=county%20court&channel=default&event=issue&amount_or_volume=10000](http://fees-register-api-prod.service.core-compute-prod.internal/fees-register/fees/lookup?service=civil%20money%20claims&jurisdiction1=civil&jurisdiction2=county%20court&channel=default&event=issue&amount_or_volume=10000)

To Be:

[http://fees-register-api-prod.service.core-compute-prod.internal:80/fees-register/fees/lookup?service=civil%20money%20claims&jurisdiction1=civil&jurisdiction2=county%20court&channel=default&event=issue&keyword=[keywords-from-the-excel]&amount_or_volume=10000](http://fees-register-api-prod.service.core-compute-prod.internal/fees-register/fees/lookup?service=civil%20money%20claims&jurisdiction1=civil&jurisdiction2=county%20court&channel=default&event=issue&keyword=keywords-from-the-excel&amount_or_volume=10000)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/caseevents/ValidateFeeCallbackHandler.java [hunk 1]
```diff
@@ -47,7 +49,9 @@ public List<CaseEvent> handledEvents() {
     private CallbackResponse validateFee(CallbackParams callbackParams) {
         var caseData = callbackParams.getCaseData();
         List<String> errors = new ArrayList<>();
-        var fee = feesService.getFeeDataByClaimValue(caseData.getClaimValue());
+        var fee = feesService.getFeeDataByClaimValue(caseData.getClaimValue(),
```
Comment (HarryH96) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/caseevents/ValidateFeeCallbackHandler.java:52]: @asthamalviya what relevance does Spec Category have on the Fee? Are these changes not for Spec and Unspec?

File: src/main/java/uk/gov/hmcts/reform/fees/client/model/ServiceTypeDto.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.time.OffsetDateTime;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+//@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
```
[the remark this answers is not in the document]
Comment (asthamalviya) [reply src/main/java/uk/gov/hmcts/reform/fees/client/model/ServiceTypeDto.java:13]: They are not used in our system but we had to copy them from latest fee and api jar without any modification.

File: src/main/java/uk/gov/hmcts/reform/fees/client/model/ApplicantTypeDto.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.time.OffsetDateTime;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+//@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
```
Comment (HarryH96) [inline src/main/java/uk/gov/hmcts/reform/fees/client/model/ApplicantTypeDto.java:13]: same again

File: src/main/java/uk/gov/hmcts/reform/fees/client/model/ChannelTypeDto.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.time.OffsetDateTime;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+//@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
```
Comment (HarryH96) [inline src/main/java/uk/gov/hmcts/reform/fees/client/model/ChannelTypeDto.java:13]: same again

File: src/main/java/uk/gov/hmcts/reform/fees/client/model/ApplicantTypeDto.java (hunk 3 above)
Comment (asthamalviya) [reply src/main/java/uk/gov/hmcts/reform/fees/client/model/ApplicantTypeDto.java:13]: Will remove the commented line

File: src/main/java/uk/gov/hmcts/reform/fees/client/model/ChannelTypeDto.java (hunk 4 above)
Comment (asthamalviya) [reply src/main/java/uk/gov/hmcts/reform/fees/client/model/ChannelTypeDto.java:13]: Will remove the commented line

**Review by HarryH96 — DISMISSED** [review]

**Review by indu2019 — DISMISSED** [review]
I've gone through the code changes with Astha and they look okay to me

**Comment by asthamalviya** [comment]
Closing because we need more clarification on requirement

**Pull request closed without merging**